tests/src/common/lease.rs

```diff
@@ -245,6 +252,24 @@ fn do_swap<DownpaymentC, LeaseC>(
         exp_lease
     };
 
+    app.send_tokens(
+        Addr::unchecked(ica_addr),
+        Addr::unchecked(ADMIN),
+        &if currency::equal::<DownpaymentC, LeaseC>() {
```

Comment:
there are three times you call `!currency::equal::<DownpaymentC, LeaseC>` in this function. What about doing that once before calling it, computing the values at one place beforehand?